.github/workflows/ci.yml

```diff
@@ -2,137 +2,25 @@ name: CI
 
-  build:
-    runs-on: ubuntu-latest
+  test:
+    name: Ruby
     needs: rubocop
-
-    services:
-      db:
-        image: postgres:13
-        ports: ['5432:5432']
-        env:
-          POSTGRES_USER: foreman
-          POSTGRES_PASSWORD: foreman
-          POSTGRES_DB: foreman
-        options: >-
-          --health-cmd pg_isready
-          --health-interval 10s
-          --health-timeout 5s
-          --health-retries 5
-    
-    steps:
-    - uses: actions/checkout@v3
-      with:
-        repository: theforeman/foreman
-        ref: develop
-        path: foreman
-    - uses: actions/checkout@v3
-      with:
-        repository: theforeman/foreman_fog_proxmox
-        path: foreman_fog_proxmox
-    - uses: actions/checkout@v3
-      with:
-        repository: fog/fog-proxmox
-        path: fog-proxmox
-    - uses: ruby/setup-ruby@v1
-      with:
-        ruby-version: '2.7'
-        bundler-cache: true
-    - name: Setup foreman
-      env:
-        PGHOST: 127.0.0.1
-        PGDB: foreman
-        PGPWD: foreman
-        PGUSER: foreman
-        RAILS_ENV: test
-        DISABLE_SPRING: true
-      working-directory: ./foreman 
-      run: |
-        sudo apt-get update && sudo apt-get -y install build-essential libpq-dev libsystemd-dev
-        gem install bundler -v 2.4.22
-        echo -n -e "gem 'foreman_fog_proxmox', :path => '../foreman_fog_proxmox'\n gem 'fog-proxmox', :path => '../fog-proxmox'\n gem 'simplecov'" > bundler.d/Gemfile.local.rb
-        cp config/settings.yaml.test config/settings.yaml
-        echo -n -e "test:\n adapter: postgresql \n port: 5432 \n database: foreman \n username: foreman \n password: foreman \n encoding: utf8 \n pool: 10" > config/database.yml
-        cp config/ignored_environments.yml.sample config/ignored_environments.yml
-        bundle config set without 'libvirt ovirt mysql2'
-        bundle install --jobs 4 --retry 3
-        bundle exec bin/rake db:migrate
-        bundle exec bin/rake db:seed
-    - uses: actions/setup-node@v1
-      with:
-        node-version: '14'
-    - name: Setup node assets
-      working-directory: ./foreman
-      env:
-        PGHOST: 127.0.0.1
-        PGDB: foreman
-        PGPWD: foreman
-        PGUSER: foreman
-        RAILS_ENV: test
-        DISABLE_SPRING: true
-      run: |
-        npm install        
-        bundle exec bin/rake assets:precompile locale:pack webpack:compile
-    - name: Run test suite with coverage report
-      working-directory: ./foreman 
-      env:
-        PGHOST: 127.0.0.1
-        PGDB: foreman
-        PGPWD: foreman
-        PGUSER: foreman
-        RAILS_ENV: test
-        DISABLE_SPRING: true
-        CC_TEST_REPORTER_ID: ${{secrets.CC_TEST_REPORTER_ID}}
-        COVERAGE: true
-      run: |
-        bundle exec bin/rake test:foreman_fog_proxmox
-    - name: Upload simplecov results for coverage
-      uses: actions/upload-artifact@v1
-      with:
-        name: coverage
-        path: foreman_fog_proxmox/coverage/
```

Comment:
I wonder how we could nicely implement that in https://github.com/theforeman/actions/

The following details need to be considered:
- we run the tests as a matrix (each branch tests something else, `test:$plugin`, `db:seed`, etc), so we'd need unique artifact names somehow
- the path each plugin wants to archive might be different

My current idea is:
- have an input `artifact_path`, which gets passed 1:1 to `actions/upload-artifact`
- only execute the step if `artifact_path` is actually defined *AND* the `task` is `test:$plugin`, so it runs only once

That would allow us to have a stable simple artifact name (`test_artifacts` or `$plugin_artifacts`) and allow other steps to consume that again.

@ekohl whatcha think?

Reply:
I usually don't care about coverage that much. I don't feel like there's a good example of it being used, but perhaps I'm wrong.

I've opened https://github.com/theforeman/actions/issues/41 and would suggest to move forward without coverage now.

Reply:
I agree that "coverage as a quality metric" is bullshit. But I do find it useful to know the piece of code I am touching is covered by tests, which means that at least in theory when I introduce unintentional behaviour changes, things get red.